### ISSUE TITLE
[Enhancement]: Fix stat issues caused compressed chunks

### ISSUE THREAD

**Issue by kgyrtkirk** [timescale/timescaledb#5548]
### What type of enhancement is this?

Performance

### What subsystems and features will be improved?

Compression

### What does the enhancement do?

Currently stats are not collected on compressed chunks ; which may lead to worse planner decisions.

<details>
<summary>Statements to evaluate actual behaviour</summary>

* setup
```
\x off
DROP TABLE IF EXISTS sensor_data;

CREATE TABLE sensor_data(
    time timestamptz NOT NULL,
    sensor_id integer NOT NULL,
    cpu double precision NULL,
    temperature double precision NULL
);

SELECT
FROM
    create_hypertable('sensor_data', 'time');

ALTER TABLE sensor_data SET (timescaledb.compress = TRUE, timescaledb.compress_orderby = 'time', timescaledb.compress_segmentby = 'sensor_id');

CREATE OR REPLACE PROCEDURE add_data(o integer)
LANGUAGE plpgsql
AS $$
BEGIN
    INSERT INTO sensor_data
    SELECT
        time +(INTERVAL '1 minute' * random()) AS time,
        sensor_id + o,
        random() AS cpu,
        random() * 100 AS temperature
    FROM
        generate_series('2018-03-02 1:00'::timestamptz, '2018-03-28 1:00', '1 hour') AS g1(time),
    generate_series(1, 100, 1) AS g2(sensor_id)
ORDER BY
    time;
END
$$;

CALL add_data(0);

-- note that: no analyze called before compressing data
SELECT
    compress_chunk(show_chunks('sensor_data'), TRUE);

ANALYZE sensor_data;

\x on
SELECT
    'problem#1: no stat data for compressed chunks; (if there were no analyze prior to compression on the uncompressed data)',
    min(stavalues1)
FROM
    pg_statistic
WHERE
    starelid = 'sensor_data'::regclass::oid
    AND staattnum = 2;

CALL add_data(1000);

ANALYZE sensor_data;

SELECT
    'problem#2 stats describes only >=1000 only; expected to also have some from 0~100',
    stavalues1
FROM
    pg_statistic
WHERE
    starelid = 'sensor_data'::regclass::oid
    AND staattnum = 2;

SELECT
    decompress_chunk(show_chunks('sensor_data'), TRUE);

CALL add_data(2000);

-- at the cost of decompressing the whole table stat can be corrected
ANALYZE sensor_data;

SELECT
    compress_chunk(show_chunks('sensor_data'), TRUE);

SELECT
    'ok: stats describes 0~2100',
    stavalues1
FROM
    pg_statistic
WHERE
    starelid = 'sensor_data'::regclass::oid
    AND staattnum = 2;

CALL add_data(3000);

ANALYZE sensor_data;

SELECT
    'problem#3: even after that - recompressing and running analyze will get back to having stats only for uncompressed data',
    stavalues1
FROM
    pg_statistic
WHERE
    starelid = 'sensor_data'::regclass::oid
    AND staattnum = 2;
```
</details>

### Implementation challenges

`ANALYZE` most likely doesn't consider the compressed rows during its operation - which makes it producing inaccurate results.

This might cause that sometimes uncompressing a single chunk of a fully compressed table produces better results - as mentioned in: #4223.

**Comment by mkindahl** [comment]
This was added in #5510 and released in 2.11.0, so closing.

**Comment by kgyrtkirk** [comment]
I don't think it was fixed by that ticket
@mkindahl did you run the validation commands in the ticket?
